src/Microsoft.AspNet.Authentication.Cookies/Events/CookieAuthenticationEvents.cs

```diff
@@ -54,30 +58,30 @@ public class CookieAuthenticationEvents : ICookieAuthenticationEvents
         /// Implements the interface method by invoking the related delegate method
         /// </summary>
         /// <param name="context"></param>
-        public virtual void ResponseSignIn(CookieResponseSignInContext context) => OnResponseSignIn(context);
+        public virtual Task ResponseSignIn(CookieResponseSignInContext context) => OnResponseSignIn(context);
```

Comment:
I actually didn't mean for any async changes to bleed into this pr yet. I will revert those. And do a new pr for those stand alone